model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -173,10 +173,19 @@ object AddressResponseAddress {
 
     val poBoxNumber = if (paf.poBoxNumber.isEmpty) "" else s"PO BOX ${paf.poBoxNumber}"
 
+    val trimmedBuildingNumber = paf.buildingNumber.trim
+    val trimmedDependentThoroughfare = paf.dependentThoroughfare.trim
+    val trimmedThoroughfare = paf.thoroughfare.trim
+
+    val buildingNumberWithStreetName = if (trimmedDependentThoroughfare.isEmpty) {
```

Comment:
Im seeing some repeated 
`.map(_.trim).filter(_.nonEmpty).mkString(", ")`
Maybe make it a def?
`def delimit(strs: String*): String = strs.map(_.trim).filter(_.nonEmpty).mkString(", ")`
The impl looks prettier and is more descriptive of what you're doing here:
`delimit(paf.departmentName, paf.organizationName, paf.subBuildingName, paf.buildingName /* .....etc */)`